nixos/modules/services/networking/pounce.nix

```diff
@@ -0,0 +1,224 @@
+          The data directory for pounce.
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.pounce;
+        defaultText = "pkgs.pounce";
+        description = ''
+          Set version of pounce package to use.
+        '';
+      };
+
+      hostAddress = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = ''
+         Where pounce should listen for incoming connections
+        '';
+      };
+
+      hostPort = mkOption {
+        type = types.int;
+        default = 6697;
+        description = ''
+        '';
+      };
+
+      sslCa = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "~/.config/pounce/pouncecert.pem";
+        description = ''
+          Path for TLS Certificate for CertFP. Needed
+          for logging to Pounce password-less.
+        '';
+      };
+
+      sslKey = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/letsencrypt/cert/private.key";
+        description = ''
+          Path to server TLS certificate key.
+        '';
+      };
+
+      sslCert = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/letsencrypt/cert/cert.crt";
+        description = ''
+           Path to server TLS certificate.
+        '';
+      };
+
+      certFP = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/pounce/libera-user.pem";
+        description = ''
+           Path to certFP certificate for login to IRC server
+        '';
+      };
+
+      # If you want to use password, you need to generate it with pounce -x
+      hashedPass = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+        '';
+      };
+
+      ircHost = mkOption {
+        type = types.str;
+        example = "irc.libera.chat";
+        default = "";
+        description = ''
+          IRC channel
+        '';
+      };
+
+      ircHostPort = mkOption {
+        type = types.int;
+        default = 6697;
+        description = ''
+        '';
+      };
+
+      joinChannels = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = ''
+          List of channel to join at startup
+        '';
+      };
+
+      ircUser = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC user
+        '';
+      };
+
+      ircNick = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC nickname
+        '';
+      };
+
+      ircPass = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC password
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = "Extra configuration";
```

Comment:
please note here where this configuration goes so users can find documentation on what to put here.

Reply:
Seems like this module should include a `settings` option instead as described by https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md.